### DISCUSSION TITLE
Problem with instalation on Windows 11 x64

### DISCUSSION THREAD

**Discussion by ManiekGrob86** [openssl/openssl#24815] [Q&A]
after perl configure i want to make nmake nad it gives me so warning

 "cl"  /Zi /Fdossl_static.pdb /Gs0 /GF /Gy /MD /W3 /wd4090 /nologo /O2 -I"." -I"include" -I"apps\include" -D"L_ENDIAN" -D"OPENSSL_PIC" -D"OPENSSLDIR=\"C:\\Program Files\\Common Files\\SSL\"" -D"ENGINESDIR=\"C:\\Program Files\\OpenSSL\\lib\\engines-3\"" -D"MODULESDIR=\"C:\\Program Files\\OpenSSL\\lib\\ossl-modules\"" -D"OPENSSL_BUILDING_OPENSSL" -D"OPENSSL_SYS_WIN32" -D"WIN32_LEAN_AND_MEAN" -D"UNICODE" -D"_UNICODE" -D"_CRT_SECURE_NO_DEPRECATE" -D"_WINSOCK_DEPRECATED_NO_WARNINGS" -D"NDEBUG"   -c /Foapps\lib\libapps-lib-app_libctx.obj "apps\lib\app_libctx.c"
app_libctx.c
C:\projects\openpace\src\openssl\include\openssl/crypto.h(25): fatal error C1083: Nie można otworzyć pliku dołącz: 'stdlib.h': No such file or directory
NMAKE : fatal error U1077: „"cl"  /Zi /Fdossl_static.pdb /Gs0 /GF /Gy /MD /W3 /wd4090 /nologo /O2 -I"." -I"include" -I"apps\include" -D"L_ENDIAN" -D"OPENSSL_PIC" -D"OPENSSLDIR=\"C:\\Program Files\\Common Files\\SSL\"" -D"ENGINESDIR=\"C:\\Program Files\\OpenSSL\\lib\\engines-3\"" -D"MODULESDIR=\"C:\\Program Files\\OpenSSL\\lib\\ossl-modules\"" -D"OPENSSL_BUILDING_OPENSSL" -D"OPENSSL_SYS_WIN32" -D"WIN32_LEAN_AND_MEAN" -D"UNICODE" -D"_UNICODE" -D"_CRT_SECURE_NO_DEPRECATE" -D"_WINSOCK_DEPRECATED_NO_WARNINGS" -D"NDEBUG"   -c /Foapps\lib\libapps-lib-app_libctx.obj "apps\lib\app_libctx.c"”: kod powrotu „0x2”
Stop.
NMAKE : fatal error U1077: „"C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.40.33807\bin\HostX86\x86\nmake.exe" /                   _build_sw”: kod powrotu „0x2”

i dont know what to do, please help me.

I use VS Community Studio 2022 and Perl strawberry

**Comment by ManiekGrob86** [comment]
i download the version openssl3.3 nad now i got something like this

Narzędzie do konserwacji programów firmy Microsoft (R) 14.40.33811.0
Copyright (C) Microsoft Corporation. Wszelkie prawa zastrzeżone.

        "cl"  /Zi /Fdossl_static.pdb /Gs0 /GF /Gy /MD /W3 /wd4090 /nologo /O2 -I"." -I"include" -I"apps\include" -D"L_ENDIAN" -D"OPENSSL_PIC" -D"OPENSSLDIR=\"C:\\Program Files\\Common Files\\SSL\"" -D"ENGINESDIR=\"C:\\Program Files\\OpenSSL\\lib\\engines-3\"" -D"MODULESDIR=\"C:\\Program Files\\OpenSSL\\lib\\ossl-modules\"" -D"OPENSSL_BUILDING_OPENSSL" -D"OPENSSL_SYS_WIN32" -D"WIN32_LEAN_AND_MEAN" -D"UNICODE" -D"_UNICODE" -D"_CRT_SECURE_NO_DEPRECATE" -D"_WINSOCK_DEPRECATED_NO_WARNINGS" -D"NDEBUG"   -c /Foapps\lib\libapps-lib-app_libctx.obj "apps\lib\app_libctx.c"
app_libctx.c
        cmd /C ""cl" /Zs /showIncludes  /Zi /Fdossl_static.pdb /Gs0 /GF /Gy /MD /W3 /wd4090 /nologo /O2 -I"." -I"include" -I"apps\include" -D"L_ENDIAN" -D"OPENSSL_PIC" -D"OPENSSLDIR=\"C:\\Program Files\\Common Files\\SSL\"" -D"ENGINESDIR=\"C:\\Program Files\\OpenSSL\\lib\\engines-3\"" -D"MODULESDIR=\"C:\\Program Files\\OpenSSL\\lib\\ossl-modules\"" -D"OPENSSL_BUILDING_OPENSSL" -D"OPENSSL_SYS_WIN32" -D"WIN32_LEAN_AND_MEAN" -D"UNICODE" -D"_UNICODE" -D"_CRT_SECURE_NO_DEPRECATE" -D"_WINSOCK_DEPRECATED_NO_WARNINGS" -D"NDEBUG"   "apps\lib\app_libctx.c" > apps\lib\libapps-lib-app_libctx.d 2>&1"
**'cmd' is not recognized as an internal or external command,
operable program or batch file.**
NMAKE : fatal error U1077: „cmd /C ""cl" /Zs /showIncludes  /Zi /Fdossl_static.pdb /Gs0 /GF /Gy /MD /W3 /wd4090 /nologo /O2 -I"." -I"include" -I"apps\include" -D"L_ENDIAN" -D"OPENSSL_PIC" -D"OPENSSLDIR=\"C:\\Program Files\\Common Files\\SSL\"" -D"ENGINESDIR=\"C:\\Program Files\\OpenSSL\\lib\\engines-3\"" -D"MODULESDIR=\"C:\\Program Files\\OpenSSL\\lib\\ossl-modules\"" -D"OPENSSL_BUILDING_OPENSSL" -D"OPENSSL_SYS_WIN32" -D"WIN32_LEAN_AND_MEAN" -D"UNICODE" -D"_UNICODE" -D"_CRT_SECURE_NO_DEPRECATE" -D"_WINSOCK_DEPRECATED_NO_WARNINGS" -D"NDEBUG"   "apps\lib\app_libctx.c" > apps\lib\libapps-lib-app_libctx.d 2>&1"”: kod powrotu „0x1”
Stop.
NMAKE : fatal error U1077: „"C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.40.33807\bin\HostX86\x86\nmake.exe" /                   _build_sw”: kod powrotu „0x2”
Stop.

**Comment by ManiekGrob86** [comment]
after given system32 to path i got now this:

cmd /C ""cl" /Zs /showIncludes  /Zi /Fdossl_static.pdb /Gs0 /GF /Gy /MD /W3 /wd4090 /nologo /O2 -I"crypto" -I"include" -I"providers\implementations\include" -I"providers\common\include" -D"L_ENDIAN" -D"OPENSSL_PIC" -D"OPENSSLDIR=\"C:\\Program Files\\Common Files\\SSL\"" -D"ENGINESDIR=\"C:\\Program Files\\OpenSSL\\lib\\engines-3\"" -D"MODULESDIR=\"C:\\Program Files\\OpenSSL\\lib\\ossl-modules\"" -D"OPENSSL_BUILDING_OPENSSL" -D"OPENSSL_SYS_WIN32" -D"WIN32_LEAN_AND_MEAN" -D"UNICODE" -D"_UNICODE" -D"_CRT_SECURE_NO_DEPRECATE" -D"_WINSOCK_DEPRECATED_NO_WARNINGS" -D"NDEBUG"   -DOPENSSL_BN_ASM_GF2m -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_CPUID_OBJ -DOPENSSL_IA32_SSE2 "ssl\record\methods\tls_pad.c" > ssl\record\methods\libcommon-lib-tls_pad.d 2>&1"
        "lib" /nologo /out:providers\libcommon.lib @C:\Users\G-TRAN~1\AppData\Local\Temp\nmF9B0.tmp
        IF EXIST .manifest DEL /F /Q .manifest
        IF EXIST libcrypto-3-x64.dll DEL /F /Q libcrypto-3-x64.dll
        cmd /C ""link" /nologo /debug /dll  /nologo /debug @C:\Users\G-TRAN~1\AppData\Local\Temp\nmFC32.tmp /implib:libcrypto.lib || (DEL /Q libcrypto-3-x64.* libcrypto.lib & EXIT 1)"
crypto\aes\libcrypto-shlib-aes_cfb.obj : fatal error LNK1112: wystąpił konflikt typu maszyny modułu „x86” z typem maszyny docelowej „x64”
Could Not Find C:\projects\openpace\src\openssl\libcrypto-3-x64.*
NMAKE : fatal error U1077: „cmd /C ""link" /nologo /debug /dll  /nologo /debug @C:\Users\G-TRAN~1\AppData\Local\Temp\nmFC32.tmp /implib:libcrypto.lib || (DEL /Q libcrypto-3-x64.* libcrypto.lib & EXIT 1)"”: kod powrotu „0x1”
Stop.
NMAKE : fatal error U1077: „"C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.40.33807\bin\HostX86\x86\nmake.exe" /                   _build_sw”: kod powrotu „0x2”
Stop.

**Reply by andrewkatson** [reply]
When I got this problem I needed an environment variable called `LIB` pointing to this directory: `C:\Program Files (x86)\Windows Kits\10\Lib\10.0.22621.0\um\x86`
